src/arch/host/include/arch/string.h

```diff
@@ -32,7 +32,28 @@
 #ifndef __INCLUDE_ARCH_STRING_SOF__
 #define __INCLUDE_ARCH_STRING_SOF__
 
+#include <errno.h>
+
 #define arch_memcpy(dest, src, size) \
 	memcpy(dest, src, size)
 
+static inline int arch_memcpy_s(void *dest,	size_t dest_size,
+								const void *src, size_t src_size)
```

Comment:
Indent looks ok in code.
Next commit in this PR will change all memcpy calls to memcpy_s and later we need to add checks in code. 
Panics should be on case by case basics.

Reply:
The checking and panics shlould be done in memcpy (since this guarantees that all callers are checked and will provide the correct context for trace macros who use __FILE__ and __LINE__. i.e.
```
#define memcpy(dest, src, size) \
#ifdef DEBUG \
    do checks and trace errors/panic \
#endif \
     do other checks and emit trace \
    arch_memcpy(dest, src, size); \
```

Reply:
memcpy is not supposed to know anything other than where to copy and what to copy.
memcpy_s does all the checks and in case of using __vec_memcpy checks in memcpy won't work since it will not be called (each time we build with xtensa)
memcpy_s should become default function for memory operations and since it has return value this could be traced for debug purposes
I can add those traces as a third step of this task but i would like to merge changes first since PR is getting big already.

Reply:
> Indent looks ok in code.

It looks bad in my editor - visual studio code, tabs with size 8.
I guess your editor just shows tabs as 4 spaces